tests/test_types.py

```diff
@@ -5369,61 +5362,103 @@ class Model(BaseModel):
 
     with pytest.raises(ValidationError) as exc_info:
         Model(lower='')
+    # insert_assert(exc_info.value.errors(include_url=False))
     assert exc_info.value.errors(include_url=False) == [
         {
-            'ctx': {'min_length': 1},
-            'input': '',
+            'type': 'too_short',
             'loc': ('lower',),
-            'msg': 'String should have at least 1 characters',
-            'type': 'string_too_short',
+            'msg': 'Value should have at least 1 item after validation, not 0',
+            'input': '',
+            'ctx': {'field_type': 'Value', 'min_length': 1, 'actual_length': 0},
+        }
+    ]
+
+
+def test_constraint_dataclass() -> None:
```

Comment:
I added this test just because I found it as part of developing this PR and fixed the bug (see changes in`_generate_schema.py`) but I could move it out to another PR if desired.